### DISCUSSION TITLE
PUT/api/{dossierID}/instances/{instanceId}/filters

### DISCUSSION THREAD

**Discussion by ChristiaanEmpowerBI** [MicroStrategy/rest-api-docs#37] [Q&A] [answered]
I have MicroStrategy 2021 Update 11 installed on my home PC, Windows 10. I only use the MicroStrategy REST API Explorer.

I am following this workflow:
docs/common-workflows/analytics/filter-data/filter-dossier-instances/apply-filters-to-a-dossier.md
I am able to get the Authentication Token, Project ID, Dossier ID, Instance ID, visualizationKey and key of the filter.

At PUT/api/{dossierID}/instances/{instanceId}/filters, I get stuck. 

My Dossier looks like this:
![image](https://github.com/MicroStrategy/rest-api-docs/assets/139224196/a39e5424-d250-4389-8b73-8f89a0e5dcee)

My expectation is that, using the REST API, I can log on to my project, create an instance to the Dossier, apply the Search Box filter to the Dossier, and extract Customer, Revenue and Revenue Derived from the Dossier.  

The request body looks like this:
![image](https://github.com/MicroStrategy/rest-api-docs/assets/139224196/6410fb84-cca9-460b-8eb7-dc0ab996d719)

Firstly, can I get documentation somewhere of what these parameters mean? 

Secondly, My best attempt gets to this:
[
  {
    "key": "W2FB6A76DE9734A3D8D9004FC482AC3B3",
    "name": "Customer",
    "selections": [
      {
        "ID": "7796",
        "Last Name": "Aaby"
      }
    ],
    "qualifier": "OR",
    "currentSelection": {
      "selectionStatus": "included"
    },
    "visualizationKey": "K52"
  }
]

But alas, I get this error, and have no idea how to troubleshoot it:
![image](https://github.com/MicroStrategy/rest-api-docs/assets/139224196/206d332d-8704-476c-9ba7-b7a51cd25723)

**Comment by ChristiaanEmpowerBI** [comment]
This issue makes me feel very dumb. I have spent 2 full days on trying to figure this out.
The best documentation I could find is this, but even this makes very little sense to me. 
Any help will be appreciated as I am a my wit's end.
![image](https://github.com/MicroStrategy/rest-api-docs/assets/139224196/43adb584-6289-460b-b95a-b51adcd57c86)

**Comment by ChristiaanEmpowerBI** [comment]
I also found this in postman:
![image](https://github.com/MicroStrategy/rest-api-docs/assets/139224196/a6cfa642-d887-4779-a280-a1187a7101f2)

but no example of how I can apply it.

**Reply by Dholmblad** [reply, marked as answer]
@ChristiaanEmpowerBI The search box filter is essentially a different view of an element list. 
Here is an example
<img width="1042" alt="Screenshot 2023-10-04 at 3 03 46 PM" src="https://github.com/MicroStrategy/rest-api-docs/assets/6351751/7fff1892-6fe8-4c9d-9e72-bc0709891fdf">
<img width="594" alt="Screenshot 2023-10-04 at 3 03 40 PM" src="https://github.com/MicroStrategy/rest-api-docs/assets/6351751/4d6408db-1a0e-4ac0-b003-be7c224ae6c0">

For your dossier & data the filter can be applied either when executing the dossier instance or afterwards with the PUT /../ filters API:

POST /dossiers/{id}/instances
`{
 
  "filters": [
    {
      "key": "{FILTER KEY}",
      "selections": [
        {
          "name": "Aaby"
        }
      ]
     }
  ],
  "persistViewState": true
}`


PUT /dossiers/{id}/instances/{instanceId}/filters

[
  {
    "key": "{FILTER KEY}",
    "selections": [
      {
        "name": "Aaby"
      }
    ]
  }
]


The problem you maybe facing is the ID being provided is the ID form, but the API is expecting the attribute elementID. Elements can be obtained from GET /api/dossiers/{dossierId}/instances/{dossierInstanceId}/elements

**Reply by ChristiaanEmpowerBI** [reply]
Thank you Dholmblad.
